packages/common/src/services/audius-backend/solana.ts

```diff
@@ -121,6 +127,15 @@ export const deriveUserBankAddress = async (
   return pubkey.toString() as SolanaWalletAddress
 }
 
+export const isTransferCheckedInstruction = (
```

Comment:
Maybe make use of [isTransferCheckedInstruction](https://github.com/solana-labs/solana-program-library/blob/a08ec5098f13fa43865492fd853c6573a956525c/token/js/src/instructions/decode.ts#L183) ?

Reply:
Good callout, I actually was looking for `decodeInstruction` and missed it in the docs. I'm going to update this to properly use the library for decoding.

Reply:
Beware that `decodeInstruction` will throw if the instruction isn't from the expected program ID

Reply:
Ah see, this is what I found lacking. Given that the program has an identifier in the first byte of data, I don't understand why there's no generalized "figure out what it is and decode it" function 🤷

Reply:
Okay circling back on this: `decodeInstruction()` _does_ determine the type and then decode. `isTransferCheckedInstruction` is a helper meant to act on an already decoded instruction. The desired behavior here is to check the type of each instruction until we find a TransferChecked, and only then do we want to decode it. So I think I will leave the helper I have in place, since there is no equivalent functionality in the SPL library.
I did, however, update it to use the same unpacking mechanism as the library (via `u8().decode()`).